### PR TITLE
Migrate display-performance-test-app to use vite

### PR DESCRIPTION
**Pull request by hl662**
Migrate from custom [react-scripts](https://github.com/imodeljs/create-react-app/tree/imodeljs/packages/react-scripts) to [vite](https://vitejs.dev/). No differences in config behavior to [display-test-app](https://github.com/iTwin/itwinjs-core/tree/master/test-apps/display-test-app).

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -30,18 +30,18 @@
...
-    "**/*.build.error.log": true,
-    "**/generated-docs": true,
-    "**/staging-docs": true,
-    "**/package-deps.json": true,
-    "**/node_modules": true,
-    "**/test-apps/**/build": true,
-    "**/out": true,
-    "**/lib": true,
-    "common/temp": true
+    "**/.nyc_output": false,
```
Comment (aruniverse) [inline .vscode/settings.json:33]: revert this file
the extensions i shared the other day is a quick way to flip these values,
but i wouldnt commit the changes it made to your pr

File: test-apps/display-performance-test-app/.gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.vite-inspect/
```
Comment (aruniverse) [inline test-apps/display-performance-test-app/.gitignore:1]: dont necessarily need to create a new .gitignore for this
couldnt you add this to the one at the root?

**Review by aruniverse — COMMENTED** [review]
there are a few references to `react-scripts` in dta and dpta, could you remove those?

could you also delete `common/config/rush/browser-approved-packages.json` and then rerun `rush update` so it will re-gen that file

**Comment by aruniverse** [comment]
@DStradley can you run the image tests and make sure we didnt break anything?

**Comment by markschlosseratbentley** [comment]
> @DStradley can you run the image tests and make sure we didnt break anything?

@DStradley and I were just discussing this. :thumbsup:

**Comment by DStradley** [comment]
> @DStradley can you run the image tests and make sure we didnt break anything?

Started test (takes ~2 hrs.)

File: test-apps/display-performance-test-app/.gitignore (hunk 2 above)
Comment (hl662) [reply test-apps/display-performance-test-app/.gitignore]: done, I also updated the .gitignore in DTA

File: .vscode/settings.json (hunk 1 above)
Comment (hl662) [reply .vscode/settings.json]: reverted

**Comment by DStradley** [comment]
Restarted Image Tests with the latest changes.

**Comment by aruniverse** [comment]
@hl662 also i think you can update `common/config/rush/pnpm-config.json`
a bunch of the `ignoreMissing` was added because of cra

**Comment by DStradley** [comment]
> Restarted Image Tests with the latest changes.

Image Tests run successfully completed.

**Comment by hl662** [comment]
> @hl662 also i think you can update `common/config/rush/pnpm-config.json` a bunch of the `ignoreMissing` was added because of cra

Updated the ignoreMissing fields in `pnpm-config.json` related to cra. Approve when ready 👍

File: common/config/rush/pnpm-config.json [hunk 3]
```diff
@@ -5,12 +5,8 @@
   "globalPeerDependencyRules": {
     "ignoreMissing": [
       "@babel/core",
```
Comment (aruniverse) [inline common/config/rush/pnpm-config.json:7]: where is this being used?

**Review by aruniverse — APPROVED** [review]

File: common/config/rush/pnpm-config.json (hunk 3 above)
Comment (hl662) [reply common/config/rush/pnpm-config.json:7]: when I ran rush update, it threw the missing peer warning in `ui/appui-abstract`

**Review by ben-polinsky — APPROVED** [review]

**Review by DStradley — APPROVED** [review]

**Pull request merged**